kinesis-video-pic/src/utils/src/Mutex.c

```diff
@@ -217,24 +205,21 @@ VOID defaultFreeMutex(MUTEX mutex)
     pthread_mutex_t* pMutex = (pthread_mutex_t*) mutex;
     pthread_mutex_destroy(pMutex);
 
-    // De-allocate the memory if it's not a well-known mutex - aka if we had allocated it previously
-    if (pMutex != &globalKvsReentrantMutex && pMutex != &globalKvsNonReentrantMutex) {
-        MEMFREE(pMutex);
-    }
+    MEMFREE(pMutex);
```

Comment:
same

Reply:
don't recommend  using a global mutex  when mutex creation failed. It can dead lock or create undesired effects, since the application assume every time a  mutex created is different., not a shared one.

Reply:
We don't even need to assert on a failure - simply clean up and return INVALID_MUTEX_HANDLE_VALUE